src/components/Editor/Hooks/useProject.js

```diff
@@ -1,35 +1,62 @@
 /* eslint-disable react-hooks/exhaustive-deps */
-import { useEffect } from 'react';
-import { useDispatch } from 'react-redux'
-import { syncProject, setProject } from '../EditorSlice'
-import { defaultPythonProject } from '../../../utils/defaultProjects';
-import { useTranslation } from 'react-i18next';
+import { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { syncProject, setProject } from "../EditorSlice";
+import { defaultPythonProject } from "../../../utils/defaultProjects";
+import { useTranslation } from "react-i18next";
 
-export const useProject = (projectIdentifier = null, locale = null, accessToken = null) => {
+export const useProject = (projectIdentifier = null, accessToken = null) => {
+  const getCachedProject = (id) =>
+    JSON.parse(localStorage.getItem(id || "project"));
+  const [cachedProject, setCachedProject] = useState(
+    getCachedProject(projectIdentifier)
+  );
+  const { i18n } = useTranslation();
+  const [currentLanguage, setCurrentLanguage] = useState(i18n.language);
   const dispatch = useDispatch();
-  const { i18n } = useTranslation()
 
-  const cachedProject = JSON.parse(localStorage.getItem(projectIdentifier || 'project'))
   const loadCachedProject = () => {
-    dispatch(setProject(cachedProject))
-  }
+    dispatch(setProject(cachedProject));
+  };
 
   useEffect(() => {
-    if (locale === i18n.language) {
-      var is_cached_saved_project = (projectIdentifier && cachedProject && cachedProject.identifier === projectIdentifier)
-      var is_cached_unsaved_project = (!projectIdentifier && cachedProject)
+    setCachedProject(getCachedProject(projectIdentifier));
+  }, [projectIdentifier]);
+
+  useEffect(() => {
+    if (i18n.language === currentLanguage) {
+      const is_cached_saved_project =
+        projectIdentifier &&
+        cachedProject &&
+        cachedProject.identifier === projectIdentifier;
+      const is_cached_unsaved_project = !projectIdentifier && cachedProject;
 
       if (is_cached_saved_project || is_cached_unsaved_project) {
-        loadCachedProject()
-        return
+        loadCachedProject();
+        setCurrentLanguage(i18n.language);
```

Comment:
I think I don't understand why this was added 🤔

Reply:
for language switching, so rather than using the `locale` url param (because that should always be the same as `i18n.language` now) `currentLanguage` is saved in state and updated when `i18n.language is updated. Hope that makes sense 🤞